src/lib/openapi/spec/project-schema.ts

```diff
@@ -55,6 +55,14 @@ export const projectSchema = {
             description:
                 '`true` if the project was favorited, otherwise `false`.',
         },
+        mode: {
+            type: 'string',
+            enum: ['open', 'protected'],
+            example: 'open',
+            nullable: true,
+            description:
+                'A mode of the project affecting what actions are possible in this project.',
+        },
```

Comment:
Out of curiosity, is there a difference between `undefined` and `null` in this case? 

The property isn't listed in the schema's required properties, so we're saying it might not be there (`undefined`), but we're also saying that it could be `null`. Is there a need for that? Isn't one or the other enough?

Reply:
Good question. Theoretically it should be either open | protected or not defined. We can probably skip null. But since I don't know full implications of the change I wanted to be on the safe side and keep it as loose as possible. Over time I'd like to make it more precise. It's easier to go this way (from loose to strict) than the other way round (from strict to loose).

Reply:
Wait, so is this a response schema or a request schema? Because whether it's easier to go from loose to strict or the other way depends on that. If it's a response schema, then yes, it's easier to narrow it down and make responses more accurate. On the other hand, "be conservative in what you send, be liberal in what you accept" says that we should know which one we're sending and use the types to enforce that.

Reply:
Read schema as there's memberCount, featureCount etc.